homeassistant/components/plugwise/gateway.py

```diff
@@ -147,9 +148,12 @@ async def async_update_data():
 async def _update_listener(hass: HomeAssistant, entry: ConfigEntry):
     """Handle options update."""
     coordinator = hass.data[DOMAIN][entry.entry_id][COORDINATOR]
-    coordinator.update_interval = timedelta(
-        seconds=entry.options.get(CONF_SCAN_INTERVAL)
-    )
+    update_interval = entry.options.get(CONF_SCAN_INTERVAL)
+    if update_interval is None:
+        api = hass.data[DOMAIN][entry.entry_id][API]
+        update_interval = DEFAULT_SCAN_INTERVAL[api.smile_type]
+
+    coordinator.update_interval = timedelta(seconds=update_interval)
```

Comment:
`seconds` is not allowed to be `None` in `timedelta`.
I hesitated about this one:

1. assert in code?
2. assert in TYPE_CHECKING?
3. dropping get (`seconds=entry.options[CONF_SCAN_INTERVAL]`)
4. replicate the code fro gateway

I used 4, but maybe 3 is cleanest?

Reply:
Most of this code needs to be refactored anyways (there is some working in progress in a PR), so doesn't matter too much imho.